### PR TITLE
Fix nodegroup get when no nodes are present

### LINKED ISSUE #4927: Cannot fetch nodegroups when DesiredCapacity > 0 but no node is registered
**What were you trying to accomplish?**
Get nodegroups info
```
eksctl get nodegroup --cluster foo
```

**What happened?**

```
$ eksctl get nodegroup --cluster foo
2022-03-10 10:34:40 [ℹ]  eksctl version 0.86.0
2022-03-10 10:34:40 [ℹ]  using region eu-west-3
Error: getting nodegroup's kubernetes version: failed to list nodes: no nodes were found
```

**How to reproduce it?**

Create a nodegroup with DesiredCaapcity > 0 then get nodegroups before the first node is registered in the cluster.
-OR-
Create a nodegroup with a feature gate not enabled on eks apiserver (ex: CustomCPUCFSQuotaPeriod).
Nodes won't be able to register into the cluster and `eksctl get nodegroup` will always fails.
```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: foo
  region: eu-west-3
nodeGroups:
  - name: ng-1
    instanceType: m5a.xlarge
    desiredCapacity: 1
    kubeletExtraConfig:
        featureGates:
            RotateKubeletServerCertificate: true
            CustomCPUCFSQuotaPeriod: true
```
**Logs**

**Anything else we need to know?**

If the desired capacity of the asg is 0, eksctl skip the retrieval of this info. It should be safe to ignore the error if there is no nodes even if the desired capacity is > 0.
https://github.com/weaveworks/eksctl/blob/39b891d01cd5c0c9577ef434721438155ecc9f67/pkg/actions/nodegroup/get.go#L169-L174

**Versions**

```
$ eksctl info
eksctl version: 0.86.0
kubectl version: v1.22.5
OS: linux
```

### PR DESCRIPTION
**Pull request by vflaux**
### Description

`kubewrapper.GetNodegroupKubernetesVersion()` returns an empty version instead of an error if no nodes were found so `get nodegroups` command will not fails in such cases.

fix #4927

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by nikimanoledaki — CHANGES_REQUESTED** [review]
I left a comment with some thoughts. Please also update the unit tests! :) 
Thanks again for opening a PR to fix this! 🎉

File: pkg/kubernetes/nodegroup.go [hunk 1]
```diff
@@ -11,14 +11,16 @@ import (
...
+
 func GetNodegroupKubernetesVersion(nodes v1.NodeInterface, ngName string) (string, error) {
 	n, err := nodes.List(context.TODO(), metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("%s=%s", api.NodeGroupNameLabel, ngName),
 	})
 	if err != nil {
 		return "", errors.Wrap(err, "failed to list nodes")
 	} else if len(n.Items) == 0 {
-		return "", errors.Wrap(errors.New("no nodes were found"), "failed to list nodes")
+		return "", errors.Wrap(errors.New(NO_NODE_FOUND), "failed to list nodes")
```
[the remark this answers is not in the document]
Comment (vflaux) [reply pkg/kubernetes/nodegroup.go]: Thanks for the suggestion, it's way simpler this way.
We don't actually need to check for the returned value as `summary.Version` is not initialized at this point and so we won't overwrite anything.

**Review by aclevername — APPROVED** [review]

**Review by Himangini — APPROVED** [review]
🎉

**Review by nikimanoledaki — APPROVED** [review]
Great work!!! Thanks again for opening this PR. 🥳 Let's merge it on Monday so that it makes next week's release!

**Pull request merged**